pyseir/deployment/demographic_mapper.py

```diff
@@ -0,0 +1,729 @@
+        """
+        Maps the age-specific covid measures predicted by the MLE model to
+        the target age distribution.
+
+        Parameters
+        ----------
+        predictions: dict
+            Contains:
+            - compartments:
+              - <compartment>: time series of population at a specific
+                               infection state (susceptible, infected,
+                               hospitalized, etc.) by demographic group
+                               simulated by MLE model. Each time series is
+                               recorded as pd.DataFrame, with dates of
+                               prediction as index.
+              name of compartment include: S - susceptible, E - exposed,
+              A - asymptomatic, I - symptomatic, HGen - in non-ICU, HICU - in
+              ICU, HVent - on ventilator, N - entire population.
+            - <measure>:
+              - <measure_unit>: time series of covid measures by demographic
+                                group predicted using the MLE model.
+                                The time series is recorded as pd.DataFrame,
+                                with dates of prediction as index.
+
+        Returns
+        -------
+          : dict
+            Contains:
+            - compartments:
+              - <compartment>: time series of population at a specific
+                               infection state (susceptible, infected,
+                               hospitalized, etc.) simulated by MLE model
+                               assuming the population has the demographic
+                               distribution of the target population. Each time
+                               series is recorded as pd.DataFrame, with dates of
+                               prediction as index.
+              name of compartment include: S - susceptible, E - exposed,
+              A - asymptomatic, I - symptomatic, HGen - in non-ICU, HICU - in ICU,
+              HVent - on ventilator, N - entire population.
+            - <measure>:
+              - <measure_unit>: time series of covid measures predicted using
+                                the MLE model and averaged over target
+                                demographic distribution (adjusted by risk
+                                modification if relative risk is specified).
+                                The time series is recorded as pd.DataFrame,
+                                with dates of prediction as index.
+        """
+        # calculate weights
+        age_bin_centers = [np.mean(tup) for tup in self.parameters['age_groups']]
+        weights = self.target_age_distribution_pdf(age_bin_centers)
+        if (weights != 1).sum() == 0:
+            logging.warning('no target age distribution is given, measure is aggregated assuming age '
+                            'distrubtion at given FIPS code')
+
+        weights /= weights.sum()
```

Comment:
If `self.target_age_distribution_pdf` is returning a `pdf`, shouldn't this have already been done inside that method?

Reply:
This would be necessary when the age groups of the mle model are not same as those that are used to generate the pdf (or more accurately pmf) function. In many cases, sum of pdf would be < 1 without the normalization. Another reason of doing this is when target age distribution pdf is not given and weights is set to array of ones (although such case should be quite rare...).